EIPS/eip-draft_request_id.md

```diff
@@ -0,0 +1,155 @@
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+The `eth` protocol defines various request and response commands that are used to exchange data between Ethereum nodes. For example, to ask a peer node for a specific set of headers, a node sends it the [`GetBlockHeaders`](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#getblockheaders-0x03) command.
+
+*Citing from the [`GetBlockHeaders` spec definition](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#getblockheaders-0x03):*
+
+>`[block: {P, B_32}, maxHeaders: P, skip: P, reverse: P in {0, 1}]`
+
+>Require peer to return a `BlockHeaders` message. Reply must contain a number of block
+headers, of rising number when `reverse` is `0`, falling when `1`, `skip` blocks apart,
+beginning at block `block` (denoted by either number or hash) in the canonical chain, and
+with at most `maxHeaders` items.
+
+The node that receives the `GetBlockHeaders` command should answer it with the [`BlockHeaders`](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#blockheaders-0x04) response command accordingly.
+
+*Citing from the [`BlockHeaders` spec definition](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#blockheaders-0x04):* 
+
+>`[blockHeader_0, blockHeader_1, ...]`
+
+>Reply to `GetBlockHeaders`. The items in the list (following the message ID) are block
+headers in the format described in the main Ethereum specification, previously asked for
+in a GetBlockHeaders message. This may validly contain no block headers if none of the
+requested block headers were found. The number of headers that can be requested in a
+single message may be subject to implementation-defined limits.
+
+Let's consider a client making many simultanous requests for `GetBlockHeaders` to one of its peers. By nature it can not be guaranteed that the expected responses arrive in the same order as they were sent. For the client to associate the incoming responses to the correct requests it has to loop through all pending requests trying to match it with the incoming response based on its contents.
+
+This can be particular tricky for responses that are ambigious such as empty responses.
+
+Now consider we change both the `GetBlockHeaders` and the `BlockHeaders` command to include a `request_id`.
+
+* `GetBlockHeaders (0x03)`
+   * **Current:** `[block: {P, B_32}, maxHeaders: P, skip: P, reverse: P in {0, 1}]`
+   * **Then**: `[request_id: P, [block: {P, B_32}, maxHeaders: P, skip: P, reverse: P in {0, 1}]]`
+* `BlockHeaders (0x04)`
+   * **Current:** `[blockHeader_0, blockHeader_1, ...]`
+   * **Then**: `[request_id: P, [blockHeader_0, blockHeader_1, ...]]`
+
+The `request_id` is a simple integer set by the client when it makes the request. On the responding side, the exact same `request_id` from the incoming request is put back into the response object.
```

Comment:
This is me nitpicking but "simple integer" isn't a term which means anything, it would be slightly more clear if this said "an integer" or, maybe, "a 64-bit integer"